### DISCUSSION TITLE
D3DX support for Direct3D9?

### DISCUSSION THREAD

**Discussion by LukasTr** [dotnet/Silk.NET#2297] [Q&A] [answered]
Hi all,

we are considering migrating our legacy application, which uses Direct3D9, to Silk.NET.

We make use of some D3DX features, for example ID3DXEffect and ID3DXFont. It looks like this is not part of Silk.NET.

Is there any easy way for us to add D3DX support? Or would that be a difficult job if we don't have experience with the Silk codebase yet?

Thanks

**Comment by Perksey** [comment, marked as answer]
That's a good question.

Adding new bindings to Silk.NET 2.X isn't really something I recommend to any external contributors, primarily because the bindings generator breaks down often, but it's not impossible. You can see how we've achieved this in the past in the `generator.json` file and its associated history. Note that a specific machine setup is needed (which may warrant a VM) documented in `documentation/runner-setup.md` as our generator expects a specific version of the MSVC headers due to it requiring a specific version of clang (which modern Visual Studio/MSVC installations no longer support).

If you do end up pursuing this, please talk to us in Discord because you *will* run into problems. The generator is quite old and dusty now, and isn't really built for a library of this scale, and we're currently doing all we can to write a new one in Silk.NET 3.0. There are also no plans in the Silk.NET team to add any new bindings to 2.X ourselves at this time.

Hope this helps.

**Comment by LukasTr** [comment]
Thank you for the thorough answer, and good luck for the development of 3.0!
